struqture-py/src/fermions/fermionic_hamiltonian_system.rs

```diff
@@ -130,4 +131,64 @@ impl FermionHamiltonianSystemWrapper {
+    ///
+    /// Raises:
+    ///     TypeError: If the input is not a struqture 2.x FermionHamiltonian.
+    ///     ValueError: Conversion failed.
+    #[staticmethod]
+    pub fn from_struqture_2(input: &Bound<PyAny>) -> PyResult<FermionHamiltonianSystemWrapper> {
+        Python::with_gil(|_| -> PyResult<FermionHamiltonianSystemWrapper> {
+            let source_serialisation_meta = input.call_method0("_get_serialisation_meta").map_err(|_| {
+                PyTypeError::new_err("Trying to use Python object as a struqture-py object that does not behave as struqture-py object. Are you sure you have the right type to all functions?".to_string())
+            })?;
+            let source_serialisation_meta: String = source_serialisation_meta.extract().map_err(|_| {
+                PyTypeError::new_err("Trying to use Python object as a struqture-py object that does not behave as struqture-py object. Are you sure you have the right type to all functions?".to_string())
+            })?;
+
+            let source_serialisation_meta: struqture_2::StruqtureSerialisationMeta = serde_json::from_str(&source_serialisation_meta).map_err(|_| {
+                PyTypeError::new_err("Trying to use Python object as a struqture-py object that does not behave as struqture-py object. Are you sure you have the right type to all functions?".to_string())
+            })?;
+
+            let target_serialisation_meta = <struqture_2::fermions::FermionHamiltonian as struqture_2::SerializationSupport>::target_serialisation_meta();
+
+            struqture_2::check_can_be_deserialised(
+                &target_serialisation_meta,
+                &source_serialisation_meta,
+            )
+            .map_err(|err| PyTypeError::new_err(err.to_string()))?;
+
+            let get_bytes = input
+                .call_method0("to_bincode")
+                .map_err(|_| PyTypeError::new_err("Serialisation failed".to_string()))?;
+            let bytes = get_bytes
+                .extract::<Vec<u8>>()
+                .map_err(|_| PyTypeError::new_err("Deserialisation failed".to_string()))?;
+            let two_import: struqture_2::fermions::FermionHamiltonian = deserialize(&bytes[..])
+                .map_err(|err| {
+                PyTypeError::new_err(format!("Type conversion failed: {}", err))
+            })?;
+            let mut fermion_system = FermionHamiltonianSystem::new(None);
+            for (key, val) in struqture_2::OperateOnDensityMatrix::iter(&two_import) {
+                let value_string = key.to_string();
+                let self_key = HermitianFermionProduct::from_str(&value_string).map_err(
+                    |_err: StruqtureError| PyValueError::new_err(
+                        "Trying to obtain struqture 1.x FermionHamiltonianSystem from struqture 2.x FermionHamiltonian. Conversion failed. Was the right type passed to all functions?".to_string()
+                ))?;
+
+                let _ = fermion_system.set(self_key, val.clone());
```

Comment:
I often see this construction where we assign the result to a variable, but it seems that this way if the result of `set` is an error we won't notice, so if we don't expect an error I would handle it with `expect`.